src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -0,0 +1,311 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace System.Formats.Tar
+{
+    // Static class containing a variety of helper methods.
+    internal static class TarHelpers
+    {
+        internal const short RecordSize = 512;
+        internal const int MaxBufferLength = 4096;
+
+        internal const int ZeroChar = 0x30;
+        internal const byte SpaceChar = 0x20;
+        internal const byte EqualsChar = 0x3d;
+        internal const byte NewLineChar = 0xa;
+
+        internal const TarFileMode DefaultMode = // 644 in octal
+            TarFileMode.UserRead | TarFileMode.UserWrite | TarFileMode.GroupRead | TarFileMode.OtherRead;
+
+        // Helps advance the stream a total number of bytes larger than int.MaxValue.
+        internal static void AdvanceStream(Stream archiveStream, long bytesToDiscard)
+        {
+            if (archiveStream.CanSeek)
+            {
+                archiveStream.Position += bytesToDiscard;
+            }
+            else if (bytesToDiscard > 0)
+            {
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(minimumLength: MaxBufferLength);
+                while (bytesToDiscard > 0)
+                {
+                    int currentLengthToRead = (int)Math.Min(MaxBufferLength, bytesToDiscard);
+                    if (archiveStream.Read(buffer.AsSpan(0, currentLengthToRead)) != currentLengthToRead)
```

Comment:
Same comment as earlier about us not assuming that Read returns as much as was requested.